fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java

```diff
@@ -33,14 +34,18 @@
 
     private final HttpClient httpClient;
     private final HttpUriRequest httpRequest;
+    private final boolean compressEntity;
 
     private final Headers headers;
     private ByteArrayOutputStream bufferedOutput;
     private boolean executed;
 
-    HttpComponentsRequest(HttpClient client, HttpUriRequest request) {
+    HttpComponentsRequest(HttpClient client, HttpUriRequest request, boolean enableContentCompression) {
         this.httpClient = client;
         this.httpRequest = request;
+        this.compressEntity = enableContentCompression &&
+                request.getFirstHeader("Content-Encoding") == null &&
```

Comment:
Could you add some reasoning here? e.g. why does the request have to be `HttpEntityEnclosingRequest` and why no `Content-Encoding` should be present?

Reply:
The reasoning is that if the caller passed an explicit "Content-Encoding" we assume the content is pre-encoded, or in case of "Content-Encoding: identity" the encoding shouldn't be touched.

Reply:
HttpEntityEnclosingRequest is to identify a request with a body (POST, PUT, PATCH)

Reply:
I can add the reasoning as comment to the code, would that be sufficient?

Reply:
simplified the code, so this is not really necessary anymore.